### PR TITLE
Pre-release merge for search v0.5.6, search-api v0.2

### PR DESCRIPTION
**Pull request by mhl10**
# Release notes - search-api-0.2, search-0.5.6

The vast majority of changes are related to search APIs, particularly the classic search API reimplementation. Most files have been touched since the last release because of [Black](https://black.readthedocs.io/en/stable/) reformatting (now part of a pre-commit hook), so it may not be useful to review all files; I have highlighted a few with comments.

Many thanks to @alefnula for his big push on the classic API reimplementation and other helpful contributions!
                                
<h2>        New Feature
</h2>
<ul>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1896'>ARXIVNG-1896</a>] -         Both new and legacy APIs should support max page size of 2,000 results
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1897'>ARXIVNG-1897</a>] -         Support for legacy API query parameter
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1898'>ARXIVNG-1898</a>] -         Implement routes module for legacy API
</li>
</ul>
    
<h2>        Task
</h2>
<ul>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2965'>ARXIVNG-2965</a>] -         update base release and fix wsgi.py
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1516'>ARXIVNG-1516</a>] -         Reimplement Atom/XML serialization for metadata API
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1518'>ARXIVNG-1518</a>] -         Update OpenAPI schema with Atom/XML examples, Accept header
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1893'>ARXIVNG-1893</a>] -         Add a `query` parameter to the JSON-based API Schema
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1895'>ARXIVNG-1895</a>] -         Add start and size to openapi schema of JSON API
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1971'>ARXIVNG-1971</a>] -         Implement boolean search in Search API
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-1972'>ARXIVNG-1972</a>] -         Support for parenthetical boolean operations and search
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2043'>ARXIVNG-2043</a>] -         Handle author affiliations in Atom search results
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2668'>ARXIVNG-2668</a>] -         Stage search API and classic API on beta.arxiv.org
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2904'>ARXIVNG-2904</a>] -         Classic arXiv API error handling
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2905'>ARXIVNG-2905</a>] -         Classic arXiv API query search_query and id_list logic
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2906'>ARXIVNG-2906</a>] -         Classic arXiv API start and max_results paging
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2907'>ARXIVNG-2907</a>] -         Classic arXiv API sort order for return results
</li>
<li>[<a href='https://arxiv-org.atlassian.net/browse/ARXIVNG-2972'>ARXIVNG-2972</a>] -         Fix the ordering issue for the classical API
</li>
</ul>

### CODE REVIEW

File: search/domain/classic_api/query_parser.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+"""
+Utility module for Classic API Query parsing.
+
+Uses lark-parser (EBNF parser) [1].
```
Comment (mhl10) [inline search/domain/classic_api/query_parser.py]: The classic API reimplementation has been updated to use lark-parser.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+- repo: https://github.com/ambv/black
```
Comment (mhl10) [inline .pre-commit-config.yaml]: New pre-commit hooks. Installation instructions are in the readme.

**Review by SBBCornell — APPROVED** [review]
I cant comment on the code, but it spins up, looks as expected, and the footer links and new colors from base are correct!

File: search/domain/classic_api/query_parser.py (hunk 1 above)
Comment (bdc34) [reply search/domain/classic_api/query_parser.py]: :+1:

File: search/domain/classic_api/query_parser.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    def empty(self, tokens: List[Token]) -> Term:
+        """Return empty term for an empty string."""
+        return Term(Field.All)
+
+    def query(self, tokens: List[Token]) -> Phrase:
+        """Query is just an expression which is a singular value."""
+        return tokens[0]  # type:ignore
+
+
+QUERY_PARSER = Lark(
```
Comment (bdc34) [inline search/domain/classic_api/query_parser.py]: :+1:

**Review by bdc34 — APPROVED** [review]
This seems good to me.

**Review by alefnula — APPROVED** [review]

**Pull request merged**